spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/procedures/IcebergProcedureRegistry.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.procedures
+
+import org.apache.iceberg.spark.{SparkCatalog, SparkSessionCatalog}
+import org.apache.spark.sql.{AnalysisException, SparkSession}
+import org.apache.spark.sql.connector.catalog.{CatalogManager, CatalogNotFoundException, CatalogPlugin, Identifier, TableCatalog}
+
+object IcebergProcedureRegistry {
+
+  def resolve(nameParts: Seq[String]): Procedure = {
+    val catalogManager = SparkSession.active.sessionState.catalogManager
+    val (catalog, ident) = resolveCatalog(catalogManager, nameParts)
+
+    validateCatalog(catalog)
+    validateProcedureIdentifier(ident)
+
+    ident.name match {
+      case "rollback_to_snapshot" => RollbackToSnapshotProcedure(catalog.asInstanceOf[TableCatalog])
```

Comment:
RollbackToSnapshotProcedure.name

Reply:
What about `rollback` instead of `rollback_to_snapshot`? We can make the target of rollback clear in the args, right?

Reply:
We will want to support a rollback [to time as well as to snapshot](https://docs.google.com/document/d/1Nf8c16R2hj4lSc-4sQg4oiUUV_F4XqZKth1woEo6TN8/).  If we want to have a single procedure, we can make `snapshot_id` and `time` as optional arguments and require at least one to be given. However, Presto already defines `rollback_to_snapshot` so I followed that.